src/ntfs/write.rs

```diff
@@ -224,181 +200,283 @@ struct Fares<'a> {
-                .or_else(|| {
-                    warn!(
-                        "Failed to find Ticket for TicketUseId {:?}",
-                        ticket_use_perimeter.ticket_use_id
-                    );
-                    None
-                })
-        });
-    for (ticket, ticket_use_perimeter) in ticket_use_perimeters {
-        let prices = get_prices(fares.ticket_prices, &ticket.id);
-        if prices.is_empty() {
-            warn!("Failed to find TicketPrice for Ticket {:?}", ticket.id);
+fn build_price_v1(id: &str, ticket: &Ticket, price: &TicketPrice) -> Result<PriceV1> {
+    // fare v1 needs prices to be integers whereas fare v2 allows floats
+    // since prices may be smaller than 1 EUR, we convert to cents, and fill fare v1 with prices in "centimes"
+    let cents_price = price.price * Decimal::from(100);
+    let cents_price = cents_price
+        .round_dp(0)
+        .to_u32()
+        .ok_or_else(|| format_err!("Cannot convert price {:?} into a u32", cents_price))?;
+    let comment = ticket.comment.clone().unwrap_or_else(String::new);
+    let price_v1 = PriceV1 {
+        id: id.to_string(),
+        start_date: price.ticket_validity_start,
+        end_date: price.ticket_validity_end + Duration::days(1), //in fare v1 end_date is excluded, whereas in fare v2 ticket_validity_end is included
+        price: cents_price,
+        name: ticket.name.clone(),
+        ignored: String::new(),
+        comment,
+        currency_type: Some("centime".to_string()),
+    };
+    Ok(price_v1)
+}
+
+fn construct_fare_v1_from_v2(fares: &Fares) -> Result<(BTreeSet<PriceV1>, BTreeSet<FareV1>)> {
+    //we check that each ticket_use_id appears only once in ticket_uses
+    check_uniqueness_of_ticket_use_ids(fares)?;
+
+    let mut prices_v1: BTreeSet<PriceV1> = BTreeSet::new();
+    let mut fares_v1: BTreeSet<FareV1> = BTreeSet::new();
+
+    // we handle ticket_use one by one
+    for ticket_use in fares.ticket_uses.values() {
+        // let's recover the included and excluded perimeters
+        // associated to our ticket_use_id
+        let (included_networks, included_lines, excluded_lines) =
+            extract_perimeter_for_ticket_use(&ticket_use.id, fares.ticket_use_perimeters)?;
+
+        if included_lines.len() + included_networks.len() == 0 {
+            warn!(
+                "The ticket_use_id {} is ignored since it has no included line or network, \
+                 and at least one must exists for a ticket_use_id to be valid.",
+                ticket_use.id
+            );
             continue;
         }
 
-        for price in prices {
-            prices_v1.insert(PriceV1::try_from((ticket, price))?);
-        }
+        // Now the restrictions for our ticket_use_id
+        let restrictions: Vec<&TicketUseRestriction> = fares
+            .ticket_use_restrictions
+            .values()
+            .filter(|restriction| restriction.ticket_use_id.as_str() == ticket_use.id)
```

Comment:
```suggestion
            .filter(|restriction| restriction.ticket_use_id == ticket_use.id)
```
No need to convert in str. Both are Strings.